engine/src/flutter/lib/web_ui/test/engine/text_editing_test.dart

```diff
@@ -1943,6 +1943,126 @@ Future<void> testMain() async {
       hideKeyboard();
     });
 
+    test('Supports composition changes with shift arrow selection in Japanese IME', () {
+      final HybridTextEditing testTextEditing = HybridTextEditing();
+      final GlobalTextEditingStrategySpy editingStrategy = GlobalTextEditingStrategySpy(
+        testTextEditing,
+      );
+
+      showKeyboard(inputType: 'text');
+      editingStrategy.composingText = 'へんかん';
+      sendFrameworkMessage(
+        codec.encodeMethodCall(
+          const MethodCall('TextInput.setEditingState', <String, dynamic>{
+            'text': 'へんかん',
+            'selectionBase': 4,
+            'selectionExtent': 4,
+            'composingBase': 0,
+            'composingExtent': 4,
+          }),
+        ),
+      );
+      checkInputEditingState(textEditing!.strategy.domElement, 'へんかん', 4, 4);
```

Comment:
In all these checks, the numbers match what was sent in `selectionBase` and `selectionExtent`. So it's not achieving the goal of the test which is to verify that selection wasn't updated on the DOM element.

To test this, you want to send some random numbers in `selectionBase` and `selectionExtent`, then check that the DOM element's selection hasn't changed to those random numbers.